src/capture.js

```diff
@@ -0,0 +1,69 @@
+import path from 'path';
+import { fileURLToPath } from 'url';
+import fs from 'fs-extra';
+import fetch from 'node-fetch';
+import admZip from 'adm-zip';
+import { createTempDir } from 'broccoli-test-helper';
+import { captureAddon } from './utils/fs-helpers.js';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+async function downloadAddon(addonPath = '') {
+  if (!addonPath.endsWith('.zip')) {
+    // assume link looks like https://github.com/NullVoxPopuli/ember-addon-migrator
+    if (addonPath.endsWith('/')) {
+      addonPath = addonPath.slice(0, -1);
+    }
+
+    addonPath = `${addonPath}/archive/refs/heads/master.zip`;
+  }
+
+  console.info(`Downloading: ${addonPath}`);
+
+  const request = await fetch(addonPath);
+  const result = await request.arrayBuffer();
+
+  const archive = new admZip(Buffer.from(result));
+  const dir = await createTempDir();
+
+  archive.extractAllTo(dir.path());
+
+  const [dirName] = fs.readdirSync(dir.path());
+
+  const dirEntry = path.join(dir.path(), dirName);
+
+  console.info(`Addon extracted to: ${dirEntry}`);
+
+  const info = fs.readJSONSync(path.join(dirEntry, 'package.json'));
+
+  return dirEntry;
+}
+
+async function capture(addonPath = '', addonName = '', addonVersion = '') {
+  if (addonPath.startsWith('http')) {
+    addonPath = await downloadAddon(addonPath);
```

Comment:
what's this for?

Reply:
@NullVoxPopuli we could use links to public addons in CI, to be able to capture or test addons we don't have commited in repo

Reply:
aslo, it's a handy way to collect list of addons to test without actually downloading it

Reply:
likely it's not needed for end users (we could have it internally)